includes/class-sensei-lesson.php

```diff
@@ -3822,6 +3822,53 @@ public function get_course_id( $lesson_id ) {
 
 	}
 
+	/**
+	 * Returns the course ID for a given set of lessons (with the key being the lesson ID)
+	 *
+	 * @since $$next-version$$
+	 * @access public
+	 *
+	 * @param array<int> $lesson_ids Array of Lesson IDs.
+	 * @return array<int, int|false> Array where the key is the lesson ID, and the value is the course ID (or false if not found)
+	 */
+	public function get_course_ids( $lesson_ids ) {
+		global $wpdb;
+
+		sort( $lesson_ids, SORT_NUMERIC );
+
+		$cache_key     = 'lesson/get-course-ids/' . implode( ',', $lesson_ids );
```

Comment:
Right now I'm not hashing on the lesson IDs part, but it's definitely something that can grow a little bit (as it can include all the lessons found for a given course, so thousands of lesson IDs)

Do you think I should apply a `sha1` or something here? I appreciate suggestions here.

Reply:
I agree that a hash would be nice. Since it doesn't have to be cryptographically secure, I think we can optimize for speed. So md5 seems like a good option to me.

Reply:
Hey people. Thanks for your feedback! Changed in [bb7a10c](https://github.com/Automattic/sensei/pull/5294/commits/bb7a10cb5caa534f38cf643d09c17ff998361d2e).